From bab51ad48824fc1794208d42eab3ae9fab27594b Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Tue, 14 Oct 2014 16:44:57 +0800
Subject: [PATCH 0012/5242] MLK-9691 net: fec: call .pm_qos_remove_request()
 in .ndo_stop() callback

commit  6d07b16041ba561d607be22f228560ddb541946e from
https://source.codeaurora.org/external/imx/linux-imx.git

Call .pm_qos_remove_request() in .ndo_stop() callback to avoid kernel
warning during enet open/close test.

Signed-off-by: Fugang Duan <B38611@freescale.com>
(cherry picked from commit: f81c176d36b8189220a729d723801d2cdd495108)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/freescale/fec_main.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index 969ca61..2d15b31 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -2967,6 +2967,7 @@ static inline bool fec_enet_irq_workaround(struct fec_enet_private *fep)
 	fec_enet_update_ethtool_stats(ndev);
 
 	fec_enet_clk_enable(ndev, false);
+	pm_qos_remove_request(&fep->pm_qos_req);
 	pinctrl_pm_select_sleep_state(&fep->pdev->dev);
 	pm_runtime_mark_last_busy(&fep->pdev->dev);
 	pm_runtime_put_autosuspend(&fep->pdev->dev);
-- 
1.7.9.5

